bee-network/bee-autopeering/src/peering/filter.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    peer::{peer_id::PeerId, Peer},
+    NeighborValidator,
+};
+
+use std::{
+    collections::HashSet,
+    sync::{Arc, RwLock, RwLockReadGuard, RwLockWriteGuard},
```

Comment:
parking_lot is faster, exact same API

Reply:
I would bench first before introducing more deps. There are many crates that do stuff more efficient than std: hashbrown, flume, parking_lot, etc, but optimizing should be done evidence based, not blindly IMHO. Also 0.1.0 doesnt need to have all bells and whistles. People just need it at all 🙂

Reply:
parking_lot is already in our dependency tree many times, including from libp2p so it's not adding any dependency. It's notoriously faster than std but I understand. We can do later.

Reply:
yea, I know it's in the dependency tree already, but my argument against it isn't binary size, but surface area or impact area of bugs (chrono crate). We shouldn't increase our surface area for dependency bugs for now compelling reason.